### PR TITLE
Rancher 311v1 ADD REINDEX ELASTICSEARCH

### PR DESCRIPTION
**Pull request by MaryiaLazavik**

### CODE REVIEW

File: pipelines/rancher/project.groovy [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!groovy
-@Library('pipelines-shared-library') _
+@Library('pipelines-shared-library@RANCHER-311V1') _
```
Comment (OHaimanov) [inline pipelines/rancher/project.groovy:2]: Before merge remove this version in lib

File: vars/jobsParameters.groovy [hunk 2]
```diff
@@ -203,6 +203,14 @@ def loadReference() {
     return _paramBoolean('load_reference', defaultTenant().loadReference, 'True if reference data should be applied')
 }
 
+def reIndexElasticsearch() {
+    return _paramBoolean('reIndexElasticsearch', true, 'True if need to reindex modules, default value: true')
```
Comment (OHaimanov) [inline vars/jobsParameters.groovy:207]: please refactor param name from camelCase to snake_case

File: vars/jobsParameters.groovy [hunk 3]
```diff
@@ -203,6 +203,14 @@ def loadReference() {
...
+}
+
+def recreateIndexElasticsearch() {
+    return _paramBoolean('recreateIndexElasticsearch', false, 'True if need to recreate index modules , default value: false')
```
Comment (OHaimanov) [inline vars/jobsParameters.groovy:211]: please refactor param name from camelCase to snake_case

File: src/org/folio/rest/Okapi.groovy [hunk 4]
```diff
@@ -144,6 +144,44 @@ class Okapi extends GeneralParameters {
...
+                    [name: 'Content-type', value: "application/json"],
+                    [name: 'X-Okapi-Tenant', value: tenant.getId()],
+                    [name: 'X-Okapi-Token', value: tenant.getAdmin_user().getToken() ? tenant.getAdmin_user().getToken() : '', maskValue: true]
+                ]
+                def resp = http.getRequest(urls, header)
+                if (resp.status == HttpURLConnection.HTTP_CREATED) {
+                    return tools.jsonParse(resp.status)
+                }
+            }
+        }
```
Comment (OHaimanov) [inline src/org/folio/rest/Okapi.groovy:182]: This should be a separate method

File: src/org/folio/rest/TenantService.groovy [hunk 5]
```diff
@@ -52,6 +52,10 @@ class TenantService extends GeneralParameters {
             permissions.assignUserPermissions(tenant, admin_user, permissions.getAllPermissions(tenant))
             users.setPatronGroup(tenant, tenant.admin_user, users.getPatronGroupId(tenant, admin_user))
             edge.createEdgeUsers(tenant, enableList)
+            if (reIndexElasticsearch) {
+                def test = okapi.reIndexElasticsearch(tenant, admin_user, recreateIndexElasticsearch)
+                return (test)
```
Comment (OHaimanov) [inline src/org/folio/rest/TenantService.groovy:57]: what is the purpose of this test return?

**Review by OHaimanov — CHANGES_REQUESTED** [review]

**Review by TarasSpashchenko — CHANGES_REQUESTED** [review]
@MaryiaLazavik please fix the comments provided by @OHaimanov

File: src/org/folio/rest/Okapi.groovy [hunk 6]
```diff
@@ -173,19 +172,30 @@ class Okapi extends GeneralParameters {
...
             [name: 'Content-type', value: "application/json"],
             [name: 'X-Okapi-Tenant', value: tenant.getId()],
             [name: 'X-Okapi-Token', value: tenant.getAdmin_user().getToken() ? tenant.getAdmin_user().getToken() : '', maskValue: true]
         ]
-        def res = http.getRequest(url, headers)
-        if (res.status == HttpURLConnection.HTTP_OK) {
-            logger.info(tools.jsonParse(res.content))
-        }else {
-            throw new AbortException("not possible check id reindex." + http.buildHttpErrorMessage(res))
+        steps.timeout(15) {
```
Comment (VitalyDemchenkoEpam) [inline src/org/folio/rest/Okapi.groovy:183]: I think 15 seconds is too low. I think 10-12 hours much better value.

File: src/org/folio/rest/Okapi.groovy [hunk 7]
```diff
@@ -173,19 +172,30 @@ class Okapi extends GeneralParameters {
...
+            while (true) {
+                def res = http.getRequest(url, headers)
+                if (res.status == HttpURLConnection.HTTP_OK) {
+                    if (tools.jsonParse(res.content).status == "Ids published") {
+                        return true
+                        logger.info("reindex successufully completed")
```
Comment (VitalyDemchenkoEpam) [inline src/org/folio/rest/Okapi.groovy:189]: unreachable code, as you have "return" on a previous line. I think "return" should be removed

File: src/org/folio/rest/Okapi.groovy [hunk 8]
```diff
@@ -173,19 +172,30 @@ class Okapi extends GeneralParameters {
...
+                        break
+                    } else if (tools.jsonParse(res.content).status != "Ids published") {
+                        logger.info("Waiting timeout, haven't status: Ids published yet." + http.buildHttpErrorMessage(res))
+                        steps.sleep(10)
+                    } else {
```
Comment (VitalyDemchenkoEpam) [inline src/org/folio/rest/Okapi.groovy:194]: looks like this "else" must be for "res.status == HttpURLConnection.HTTP_OK" if.

**Review by VitalyDemchenkoEpam — APPROVED** [review]

File: src/org/folio/rest/Okapi.groovy [hunk 9]
```diff
@@ -144,6 +144,62 @@ class Okapi extends GeneralParameters {
         }
     }
 
+    /**
+     * reindex Elasticsearch to modules for tenant
+     * @param tenant
+     * @param okapiUrl
+     */
+    def reindex_elastic_search(tenant, admin_user, recreate_index_elastic_search) {
```
Comment (OHaimanov) [inline src/org/folio/rest/Okapi.groovy:152]: Please rename method to camelCase

**Review by OHaimanov — CHANGES_REQUESTED** [review]

**Review by Volodymyr-Kartsev — APPROVED** [review]

**Review by OHaimanov — APPROVED** [review]

**Pull request merged**